packages/vaex-core/vaex/dataframe.py

```diff
@@ -4000,19 +3999,31 @@ def split(self, frac):
         [2 3 4]
         [5 6 7 8 9]
 
-        :param int/list frac: If int will split the DataFrame in two portions, the first of which will have size as specified by this parameter. If list, the generator will generate as many portions as elements in the list, where each element defines the relative fraction of that portion.
+        :param int, float or list of floats into: If float will split the DataFrame in two, the first of which will have a relative length as specified by this parameter.
+            When a list, will split into as many portions as elements in the list, where each element defines the relative length of that portion.
+            When an int, split DataFrame into n dataframes of equal length (last one may deviate), if len(df) < n, it will return len(df) DataFrames.
         :return: A list of DataFrames.
         :rtype: list
         '''
         self = self.extract()
-        if _issequence(frac):
+        if isinstance(into, numbers.Integral):
+            step = max(1, vaex.utils.div_ceil(len(self), into))
+            i1 = 0
+            i2 = step
+            while i1 < len(self):
+                i2 = min(len(self), i2)
+                yield self[i1:i2]
+                i1, i2 = i2, i2 + step
+            return
+
+        if _issequence(into):
             # make sure it is normalized
-            total = sum(frac)
-            frac = [k / total for k in frac]
+            total = sum(into)
+            into = [k / total for k in into]
```

Comment:
If the user specifies a list of floats, the some of which is not 1, should we raise a warning notifying the user that the relative fractions will be normalized? 

I was expecting an error there, when I entered bunch of numbers whose sum was much greater than 1. 

(or just flat out ask for a sum of 1?).

Reply:
no, i think it's convenient like this.

Reply:
it is convenient indeed, but i think a warning of the re-normalization (if it happens) is not a bad idea.

Reply:
but how to get rid of the warning then? I mean, I think it's expected behavior.